modules/littlefs/zephyr_lfs_config.h

```diff
@@ -147,21 +147,9 @@ static inline int lfs_scmp(uint32_t a, uint32_t b)
 /* Convert between 32-bit little-endian and native order */
 static inline uint32_t lfs_fromle32(uint32_t a)
 {
-#if !defined(LFS_NO_INTRINSICS) && (                                    \
-	(defined(BYTE_ORDER) && defined(ORDER_LITTLE_ENDIAN) &&         \
-	BYTE_ORDER == ORDER_LITTLE_ENDIAN) ||                           \
-	(defined(__BYTE_ORDER) && defined(__ORDER_LITTLE_ENDIAN) &&     \
-	__BYTE_ORDER   == __ORDER_LITTLE_ENDIAN) ||                     \
-	(defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__) && \
-	__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__))
+#if !defined(CONFIG_BIG_ENDIAN)
 	return a;
-#elif !defined(LFS_NO_INTRINSICS) && (                               \
-	(defined(BYTE_ORDER) && defined(ORDER_BIG_ENDIAN) &&         \
-	BYTE_ORDER == ORDER_BIG_ENDIAN) ||                           \
-	(defined(__BYTE_ORDER) && defined(__ORDER_BIG_ENDIAN) &&     \
-	__BYTE_ORDER == __ORDER_BIG_ENDIAN) ||                       \
-	(defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) && \
-	__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__))
+#elif !defined(LFS_NO_INTRINSICS)
```

Comment:
`&& defined(CONFIG_BIG_ENDIAN)`?

Reply:
Hmm no, because it´s an `#elif`?

Reply:
Before:

```
if !LFS_NO_INTRINSICS && !CONFIG_BIG_ENDIAN
  return a;
elif !LFS_NO_INTRINSICS && CONFIG_BIG_ENDIAN
  return __builtin_bswap32(a);
else
  ...
```

Now:

```
if !CONFIG_BIG_ENDIAN
  return a;
elif !LFS_NO_INTRINSICS
  return __builtin_bswap32(a);
else:
 ...
```

If we now have: `LFS_NO_INTRINSICS && !CONFIG_BIG_ENDIAN`, we'll `return a`, whereas before the `else` branch would have been taken, and I think that is what is desired according to the description:

```c
// Builtin functions, these may be replaced by more efficient
// toolchain-specific implementations. LFS_NO_INTRINSICS falls back to a more
// expensive basic C implementation for debugging purposes
```

Reply:
It looks like the previous behavior was actually wrong, so please add a first commit fixing the issue.